app/controllers/pictures_controller.rb

```diff
@@ -0,0 +1,83 @@
+class PicturesController < ApplicationController
+  before_action :authenticate_user!, except: [:show, :index]
+  before_action :set_picture, only: [:show, :edit, :update, :destroy]
+  before_action :picture_owner, only: [:edit, :update, :destroy]
+
+def picture_owner
+  unless @picture.user_id == current_user.id
+    flash[:notice] = 'Access denied as you are not owner of this Comment'
+    redirect_to pictures_path
+  end
+end
+
+  # GET /pictures
+  # GET /pictures.json
+  def index
+    @pictures = Picture.all
+  end
+
+  # GET /pictures/1
+  # GET /pictures/1.json
+  def show
+  end
+
+  # GET /pictures/new
+  def new
+    @picture = Picture.new
+  end
+
+  # GET /pictures/1/edit
+  def edit
+  end
+
+  # POST /pictures
+  # POST /pictures.json
+  def create
+    @picture = Picture.new(picture_params)
```

Comment:
I am guessing this needs another approach instead of using 'hidden field' to ensure a nefarious coder can't replace the value I am getting from my params with something else. If I somehow fed the user id from params in the controller rather than on the html page, could this be a method to solve this? That way, the logic isn't on the page for someone to edit, but hidden in the model instead?

Reply:
Totally! The controller gets access to a number of pieces of information, not just `params`. Look at where you got the user id from to put in the form, and see if you can use a similar technique in the controller. To understand how it works, figure out how your application is keeping track of that information.